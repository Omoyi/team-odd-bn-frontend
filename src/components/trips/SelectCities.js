import React from 'react';
import '../../assets/css/base.scss';

const inputsValues = {};
const SelectCities = ({ myData, handleChange, inputName }) => inputName.map((a) => (
  <div className={a.className} key={a.className}>
    <select
      onChange={(e) => {
        inputsValues[a.name] = e.target.value;
        return handleChange(inputsValues);
      }}
      defaultValue=""
      required
    >
      <option value="" disabled>
        { a.PlaceHolderName}
      </option>

      {myData
      && myData.map((cities) => (
        <option name={a.name} value={cities.id} key={cities.id}>
          {cities.city}
        </option>
      ))}
    </select>
  </div>
));

export default SelectCities;
