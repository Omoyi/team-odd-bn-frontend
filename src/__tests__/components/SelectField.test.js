import React from 'react';
import { shallow } from 'enzyme';
import SelectCities from '../../components/trips/SelectCities';
import selectInputs from '../../constants/selectCities';

describe('First React component test with Enzyme', () => {
  it('should renders Select Input component', () => {
    expect(shallow(<SelectCities inputName={selectInputs} />)).toMatchSnapshot();
    const mockFn = jest.fn();
    const input = shallow(
      <SelectCities
        inputName={[{
          className: 'originCity', name: 'originId', PlaceHolderName: 'Your Origin', value: 'originId',
        }]}
        handleChange={mockFn}
      />,
    );
    const cc = input.find('select');
    cc.simulate('change', { target: { value: 'matched' } });
    expect(mockFn.mock.calls[0][0]).toEqual({ originId: 'matched' });
  });
});
