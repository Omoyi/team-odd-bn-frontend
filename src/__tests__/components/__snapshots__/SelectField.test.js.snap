// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First React component test with Enzyme should renders Select Input component 1`] = `ShallowWrapper {}`;
